internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go

```diff
@@ -0,0 +1,194 @@
+						Description: "The system key encrypted.",
+						Computed:    true,
+						Optional:    false,
+					},
+					"key_data": schema.StringAttribute{
+						Description: "The clear text system key base 64 encoded. The system key must be 32 bytes before base 64 encoding",
+						Computed:    true,
+						Optional:    false,
+					},
+				},
+			},
+			"previous": schema.SingleNestedAttribute{
+				Description: "Previous SystemKeys Secrets that are used in cryptographic operations to generate and consume internal tokens.",
+				Computed:    true,
+				Optional:    false,
+				Attributes: map[string]schema.Attribute{
+					"creation_date": schema.StringAttribute{
+						Description: "Creation time of the key.",
+						Computed:    true,
+						Optional:    false,
+					},
+					"encrypted_key_data": schema.StringAttribute{
+						Description: "The system key encrypted.",
+						Computed:    true,
+						Optional:    false,
+					},
+					"key_data": schema.StringAttribute{
+						Description: "The clear text system key base 64 encoded. The system key must be 32 bytes before base 64 encoding",
+						Computed:    true,
+						Optional:    false,
+					},
+				},
+			},
+			"pending": schema.SingleNestedAttribute{
+				Description: "Pending SystemKeys Secrets that are used in cryptographic operations to generate and consume internal tokens.",
+				Computed:    true,
+				Optional:    false,
+				Attributes: map[string]schema.Attribute{
+					"creation_date": schema.StringAttribute{
+						Description: "Creation time of the key.",
+						Computed:    true,
+						Optional:    false,
+					},
+					"encrypted_key_data": schema.StringAttribute{
+						Description: "The system key encrypted.",
+						Computed:    true,
+						Optional:    false,
+					},
+					"key_data": schema.StringAttribute{
+						Description: "The clear text system key base 64 encoded. The system key must be 32 bytes before base 64 encoding",
+						Computed:    true,
+						Optional:    false,
+					},
+				},
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+// Metadata returns the resource type name.
+func (r *serverSettingsSystemKeysDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_server_settings_system_keys"
+}
+
+func (r *serverSettingsSystemKeysDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readServerSettingsSystemKeysDataSource(ctx context.Context, r *client.SystemKeys, state *serverSettingsSystemKeysDataSourceModel, existingId *string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	state.Id = id.GenerateUUIDToState(existingId)
```

Comment:
This is a singleton data source, so we can use a hardcoded id here. There's no way to track an existing id with data sources.

Reply:
ah, copy/paste for a fix.  good catch